src/test_encode_decode.rs

```diff
@@ -105,22 +105,47 @@ fn setup_encoder(
 static DIMENSION_OFFSETS: &[(usize, usize)] =
   &[(0, 0), (4, 4), (8, 8), (16, 16)];
 
-#[test]
-#[ignore]
-fn speed() {
+fn speed(s: usize) {
   let quantizer = 100;
   let limit = 5;
   let w = 64;
   let h = 80;
 
   for b in DIMENSION_OFFSETS.iter() {
-    for s in 0..10 {
       encode_decode(w + b.0, h + b.1, s, quantizer, limit, 8, 15, 15, true);
-    }
   }
 }
 
-static DIMENSIONS: &[(usize, usize)] = &[
+macro_rules! test_speeds {
+  ($($S:expr),+) => {
+    $(
+        paste::item!{
+            #[test]
+            fn [<speed_ $S>]() {
+                speed($S)
+            }
+        }
+    )*
+  }
+}
+
+test_speeds!{ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
```

Comment:
Whitespace is a little odd here.
```suggestion
test_speeds!{ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }
```

Reply:
Folded in the rework.